src/main/java/org/opensearch/knn/index/vectorvalues/KNNVectorValuesIterator.java

```diff
@@ -71,18 +71,28 @@ public interface KNNVectorValuesIterator {
      * {@link DocIdSetIterator} interface. Example: {@link BinaryDocValues}, {@link FloatVectorValues} etc.
      */
     class DocIdsIteratorValues implements KNNVectorValuesIterator {
-        protected DocIdSetIterator docIdSetIterator;
-        private static final List<Function<DocIdSetIterator, Boolean>> VALID_ITERATOR_INSTANCE = List.of(
-            (itr) -> itr instanceof BinaryDocValues,
-            (itr) -> itr instanceof FloatVectorValues,
-            (itr) -> itr instanceof ByteVectorValues
-        );
-
-        DocIdsIteratorValues(@NonNull final DocIdSetIterator docIdSetIterator) {
-            validateIteratorType(docIdSetIterator);
+        private final DocIdSetIterator docIdSetIterator;
```

Comment:
I think this class will also need a bit of refactoring, for now to unblock main this is fine. Lets have a GH issue for this to fix the abstraction. Because this class otherwise is not going to be maintainable in future